src/runtime/components/IonAnimation.vue

```diff
@@ -0,0 +1,145 @@
+<script setup lang="ts">
+import { onMounted, onBeforeUnmount } from 'vue'
+import {
+  createAnimation,
+  Animation,
+  AnimationDirection,
+  AnimationFill,
+  AnimationKeyFrames,
+} from '@ionic/vue'
+
+interface AnimationFromObject {
+  property: string
+  fromValue: string
+}
+
+interface AnimationFromToObject {
+  property: string
+  fromValue: string
+  toValue: string
+}
+
+type AnimationStyles = {
+  [key: string]: any
+}
+
+interface AnimationOptions {
+  id?: string
+  duration?: number
+  iterations?: number
+  easing?: string
+  fill?: AnimationFill
+  direction?: AnimationDirection
+  from?: AnimationFromObject | AnimationFromObject[]
+  fromTo?: AnimationFromToObject | AnimationFromToObject[]
+  keyframes?: AnimationKeyFrames
+  playOnMount?: boolean
+  playOnVisible?: boolean
+  beforeStyles?: AnimationStyles
+  beforeAddClass?: string | string[]
+  beforeClearStyles?: string[]
+  afterStyles?: AnimationStyles
+  afterAddClass?: string | string[]
+  afterClearStyles?: string[]
+}
+
+const props = withDefaults(defineProps<AnimationOptions>(), {
+  id: '',
```

Comment:
is this a required prop? What happens if it isn't passed?

Reply:
I don't think it that much, as far as I know it only shows it in Chrome DevTools under that name, it just helps with [debugging](https://ionicframework.com/docs/utilities/animations#debugging)